temporalcli/commands.workflow_reset_test.go

```diff
@@ -207,3 +214,283 @@ func (s *SharedServerSuite) TestWorkflow_Reset_ToEventID() {
 	s.Equal(1, oneExecutions, "Should not have re-executed the first activity")
 	s.Equal(2, twoExecutions, "Should have re-executed the second activity")
 }
+
+func (s *SharedServerSuite) TestBatchResetByBuildId() {
+	sut := newSystemUnderTest(s)
+
+	sut.startWorkerFor(originalWorkflow, workflowOptions{name: "wf", version: "v1"})
+	sut.executeWorkflow("wf")
+	sut.waitWorkflowBlockedAfterFirstActivity()
+	sut.stopWorkerFor("v1")
+
+	sut.startWorkerFor(extendedWorkflowWithBuggyActivity, workflowOptions{name: "wf", version: "v2"})
+	sut.allowWorkflowToContinue()
+	sut.waitBadActivityExecuted()
+	sut.stopWorkerFor("v2")
+
+	sut.startWorkerFor(extendedWorkflowWithNonDeterministicFix, workflowOptions{name: "wf", version: "v3"})
+	sut.waitBlockOnNonDeterministicError()
+
+	query := fmt.Sprintf("WorkflowId = \"%s\"", sut.run.GetID())
+	s.CommandHarness.Stdin.WriteString("y\n")
+	res := s.Execute(
+		"workflow", "reset",
+		"--address", s.Address(),
+		"--reason", "test-reset-event-id",
+		"--type", "BuildId",
+		"--build-id", sut.buildPrefix+"v2",
+		"--query", query,
+	)
+	require.NoError(s.T(), res.Err)
+
+	sut.assertWorkflowComplete()
+	sut.assertOnlySecondActivityRetried()
+	sut.stopWorkerFor("v3")
+}
+
+const (
+	badActivity = iota
+	firstActivity
+	secondActivity
+	thirdActivity
+)
+
+type assertions interface {
+	NoError(error, ...interface{})
+	Error(error, ...interface{})
+	True(bool, ...interface{})
+	Eventually(func() bool, time.Duration, time.Duration, ...interface{})
+}
+
+type batchResetTestData struct {
+	t           *testing.T
+	buildPrefix string
+	assert      assertions
+	client      client.Client
+	tq          string
+	counters    []atomic.Int32
```

Comment:
```suggestion
	counters    []*atomic.Int32
```

It makes more sense to pass around pointers to this

Reply:
No, you do not need a pointer here. The protocol for batchResetTestData is by sharing already. I should have return pointer from newSystemUnderTest to make it clearer. Will fix it.

Reply:
Correct you do not _need_ it if you're sure you'll never `append`, but it's just a common pattern for atomic values to be used and passed around by pointers. Too many times people create slices for structs only to later `append` and be surprised when the underlying array when they could have just used pointers in a reasonable way when all operations require the pointer receiver. So in general in Go it's bad practice to take a pointer to a indexed slice value since a slice's underlying array can be mutated (even if it technically works here).